packages/medusa/src/services/line-item.ts

```diff
@@ -363,6 +364,56 @@ class LineItemService extends BaseService {
 
     return itemTaxLineRepo.create(args)
   }
+
+  async clone(
+    ids: string | string[],
+    data: { order_edit_id?: string; order_id?: string | null } = {},
+    options: { setOriginalLineItemId?: boolean } = {}
+  ): Promise<LineItem[]> {
+    ids = typeof ids === "string" ? [ids] : ids
+    return await this.atomicPhase_(async (manager) => {
+      let lineItems: DeepPartial<LineItem>[] = await this.list(
+        {
+          id: In(ids as string[]),
+        },
+        {
+          relations: ["tax_lines", "adjustments"],
+        }
+      )
+
+      const lineItemRepository = manager.getCustomRepository(
+        this.lineItemRepository_
+      )
+
+      lineItems = lineItems.map(
+        (item) =>
+          ({
+            ...item,
```

Comment:
todo: remove relation to `cart_id`, `swap_id`, `claim_order_id`; the goal here is only to clone the "contents" of the line item not the owner if that makes sense?

Maybe we should change this method to be more like this:

```
cloneTo(lineItemId, destinationId, data) {
  // retrieve and "clone" lineItemId
  // associate with destinationId (maybe use prefix to determine type e.g., `swap_` -> swap_id: destinationId, `oe_` -> order_edit_id: destinationId, etc.)
  // override anything from data; e.g., quantity: data.quantity
}
```

Reply:
I ve done a proposal [here](https://github.com/medusajs/medusa/pull/2255/commits/4a88de234a215daaa182de391361340dec0bad28) let me know wyt